src/java/org/apache/cassandra/dht/Token.java

```diff
@@ -75,6 +76,15 @@ public ByteSource asComparableBytes(Token token, ByteComparable.Version version)
          */
         public abstract Token fromComparableBytes(ByteSource.Peekable comparableBytes, ByteComparable.Version version);
 
+        public void skipComparableBytes(ByteSource.Peekable comparableBytes, ByteComparable.Version version, IPartitioner partitioner)
+        {
+            if (partitioner.isFixedLength())
+                ByteSourceInverse.skipBytes(comparableBytes, partitioner.getMaxTokenSize());
```

Comment:
As there is no way to understand if this is correct by looking at the surrounding code, this method should not be implemented here. It should be in the individual token classes where `asComparableBytes` and `fromComparableBytes` are implemented.

Reply:
actually I was inspired by org.apache.cassandra.dht.Token.TokenFactory#skip(DataInputPlus in, IPartitioner p) logic, I can place it next to it as an option...
Moving the logic down to the individual token classes makes sense for me as well.